rasa/nlu/utils/spacy_utils.py

```diff
@@ -129,18 +130,109 @@ def get_text(self, example, attribute):
 
         return self.preprocess_text(example.get(attribute))
 
+    @staticmethod
+    def merge_content_lists(indexed_training_samples: List, doc_lists: List) -> List:
+        """ Merge lists with processed Docs back into their original order. """
+
+        dct = dict(indexed_training_samples)
+        dct.update(dict(doc_lists))
+        return list(dct.items())
+
+    @staticmethod
+    def filter_training_samples_by_content(indexed_training_samples: List) -> Tuple:
+        """ Separates empty training samples from content bearing ones. """
+
+        docs_to_pipe = list(
+            filter(
+                lambda training_sample: training_sample[1] != "",
+                indexed_training_samples,
+            )
+        )
+        empty_docs = list(
+            filter(
+                lambda training_sample: training_sample[1] == "",
+                indexed_training_samples,
+            )
+        )
+        return docs_to_pipe, empty_docs
+
+    def process_content_bearing_samples(self, samples_to_pipe: Tuple) -> List:
+        """ Sends content bearing training samples to spaCy's pipe. """
+
+        docs = [
+            (to_pipe_sample[0], doc)
+            for to_pipe_sample, doc in zip(
+                samples_to_pipe,
+                [
+                    doc
+                    for doc in self.nlp.pipe(
+                        [txt for _, txt in samples_to_pipe], batch_size=50
+                    )
+                ],
+            )
+        ]
+        return docs
+
+    def process_non_content_bearing_samples(self, empty_samples: Tuple) -> List:
+        """ Creates empty Doc-objects from zero-lenghted training samples strings. """
+
+        n_docs = [
+            (empty_sample[0], doc)
+            for empty_sample, doc in zip(
+                empty_samples, [Doc(self.nlp.vocab) for doc in empty_samples]
+            )
+        ]
+        return n_docs
+
+    @staticmethod
+    def check_non_content_bearing_sample_order(
```

Comment:
Don't need this over here.

Reply:
Removed in [e583d48](https://github.com/RasaHQ/rasa/pull/4483/commits/e583d484f809875e73c593a6e99b21724accff2f).